agent/agent.go

```diff
@@ -295,10 +295,20 @@ func (a *Agent) prepareNextTask(ctx context.Context, nextTask *apimodels.NextTas
 func nextTaskHasDifferentTaskGroupOrBuild(nextTask *apimodels.NextTaskResponse, tc *taskContext) bool {
 	if tc.taskConfig == nil ||
 		nextTask.TaskGroup == "" ||
-		nextTask.TaskGroup != tc.taskGroup ||
 		nextTask.Build != tc.taskConfig.Task.BuildId {
 		return true
 	}
+	if nextTask.TaskGroup != tc.taskGroup {
+		grip.DebugWhen(nextTask.TaskGroup == tc.taskConfig.Task.TaskGroup, message.Fields{
```

Comment:
Is there a logger available at this point to log to, from the previous or next task? This will only end up in the local filesystem.